src/apps/yield-yak/helpers/yield-yak.vault.token-definitions-resolver.ts

```diff
@@ -28,17 +32,62 @@ const query = gql`
   }
 `;
 
+export type YieldYakFarmDetails = {
+  address: string;
+  deployed: number;
+  name: string;
+  depositToken: {
+    address: string;
+    symbol: string;
+    decimals: number;
+  };
+  totalDeposits: string;
+};
+
 @Injectable()
 export class YieldYakVaultTokenDefinitionsResolver {
-  constructor(@Inject(APP_TOOLKIT) protected readonly appToolkit: IAppToolkit) {}
+  constructor(
+    @Inject(APP_TOOLKIT) protected readonly appToolkit: IAppToolkit,
+    @Inject(YieldYakContractFactory) private readonly yieldyakContractFactory: YieldYakContractFactory,
+  ) {}
 
   @Cache({
     key: network => `studio:yield-yak:${network}:vault-data`,
     ttl: 5 * 60, // 60 minutes
   })
-  async getVaultDefinitionsData() {
-    const endpoint = 'https://api.thegraph.com/subgraphs/name/yieldyak/reinvest-tracker';
-    const data = await this.appToolkit.helpers.theGraphHelper.request<FarmsResponse>({ endpoint, query });
-    return data.farms;
+  async getVaultDefinitionsData(network: Network): Promise<FarmsResponse['farms']> {
+    const farms = await Axios.get<YieldYakFarmDetails[]>('https://staging-api.yieldyak.com/farms').then(x => x.data);
+    const multicall = this.appToolkit.getMulticall(network);
+
+    const response = await Promise.all(
+      farms.map(async farm => {
+        const farmContract = this.yieldyakContractFactory.yieldYakVault({
+          address: farm.address,
+          network: Network.AVALANCHE_MAINNET,
+        });
+
+        try {
+          const [depositTokenBalance, name, depositToken] = await Promise.all([
```

Comment:
Just went over the code in the vault token fetcher -- I suggest you removing all those calls to multicall (to get `depositTokenBalance`, `name`, and `depositToken`), as we only need the `depositToken` to be used in the get underlying token step.

It would be greatly beneficial to use `multicall` in the `getUnderlyingTokenAddresses` directly, using `multicall` passed in the fn params. Doing it this way helps our system be efficient with all the on-chain calls of the whole fetcher.

So to summarize, you can simply return an array of all the farm addresses here in the definitions resolver, and in the fetcher, do the on-chain call to get the `depositToken` in the `getUnderlyingTokenAddresses` using `multicall` from its context 😄